File: k-kbk/3주차/11723.py
import sys

input = sys.stdin.readline

m = int(input())
s = set()

for i in range(m):
    line = input().rstrip()

    if line == "all":
        s = set(range(1, 21))
        continue
    if line == "empty":
        s = set()
        continue

    command, n = line.split(" ")
    n = int(n)

    if command == "add":
        s.add(n)
        continue
    if command == "remove":
        s.discard(n)
        continue
    if command == "check":
        print(1 if n in s else 0)
        continue
    if command == "toggle":
        if n in s:
            s.remove(n)
        else:
            s.add(n)
        continue

File: keemsebin/3주차/11723.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  // 11723
  const N = parseInt(input[0]);
  const set = new Set();
  let answer = [];

  for (let i = 0; i < N; i++) {
    const [command, x] = input[i + 1].split(" ");
    let num = Number(x);
    switch (command) {
      case "add":
        set.add(num);
        break;
      case "remove":
        set.delete(num);
        break;
      case "check":
        answer.push(set.has(num) ? 1 : 0);
        break;
      case "toggle":
        if (set.has(num)) {
          set.delete(num);
        } else {
          set.add(num);
        }
        break;
      case "all":
        for (let j = 1; j <= 20; j++) {
          set.add(j);
        }
        break;
      case "empty":
        set.clear();
        break;
    }
  }
  console.log(answer.join("\n"));
});
